Jellyfin.Server/Migrations/Routines/UpdateDatabaseToTenDotTen.cs

```diff
@@ -0,0 +1,376 @@
+
+            // covering index for getitemvalues
+            "create index if not exists idx_TypeTopParentIdGuid on TypedBaseItems(Type,TopParentId,Guid)",
+
+            // used by movie suggestions
+            "create index if not exists idx_TypeTopParentIdGroup on TypedBaseItems(Type,TopParentId,PresentationUniqueKey)",
+            "create index if not exists idx_TypeTopParentId5 on TypedBaseItems(TopParentId,IsVirtualItem)",
+
+            // latest items
+            "create index if not exists idx_TypeTopParentId9 on TypedBaseItems(TopParentId,Type,IsVirtualItem,PresentationUniqueKey,DateCreated)",
+            "create index if not exists idx_TypeTopParentId8 on TypedBaseItems(TopParentId,IsFolder,IsVirtualItem,PresentationUniqueKey,DateCreated)",
+
+            // resume
+            "create index if not exists idx_TypeTopParentId7 on TypedBaseItems(TopParentId,MediaType,IsVirtualItem,PresentationUniqueKey)",
+
+            // items by name
+            "create index if not exists idx_ItemValues6 on ItemValues(ItemId,Type,CleanValue)",
+            "create index if not exists idx_ItemValues7 on ItemValues(Type,CleanValue,ItemId)",
+
+            // Used to update inherited tags
+            "create index if not exists idx_ItemValues8 on ItemValues(Type, ItemId, Value)",
+
+            "CREATE INDEX IF NOT EXISTS idx_TypedBaseItemsUserDataKeyType ON TypedBaseItems(UserDataKey, Type)",
+            "CREATE INDEX IF NOT EXISTS idx_PeopleNameListOrder ON People(Name, ListOrder)"
+        ];
+
+        string[] userDataQUeries =
+        [
+            "create table if not exists UserDatas (key nvarchar not null, userId INT not null, rating float null, played bit not null, playCount int not null, isFavorite bit not null, playbackPositionTicks bigint not null, lastPlayedDate datetime null, AudioStreamIndex INT, SubtitleStreamIndex INT)",
+            "drop index if exists idx_userdata",
+            "drop index if exists idx_userdata1",
+            "drop index if exists idx_userdata2",
+            "drop index if exists userdataindex1",
+            "drop index if exists userdataindex",
+            "drop index if exists userdataindex3",
+            "drop index if exists userdataindex4",
+            "create unique index if not exists UserDatasIndex1 on UserDatas (key, userId)",
+            "create index if not exists UserDatasIndex2 on UserDatas (key, userId, played)",
+            "create index if not exists UserDatasIndex3 on UserDatas (key, userId, playbackPositionTicks)",
+            "create index if not exists UserDatasIndex4 on UserDatas (key, userId, isFavorite)",
+            "create index if not exists UserDatasIndex5 on UserDatas (key, userId, lastPlayedDate)"
+        ];
+
+        _logger.LogInformation("Prepare database for EFCore migration: update to version 10.10.z");
+
+        var dataPath = _paths.DataPath;
+
```

Comment:
it is common for library.db migrations to backup the db before modifications are done